karapace/schema_reader.py

```diff
@@ -254,105 +266,122 @@ def handle_messages(self):
-        elif key["keytype"] == "SCHEMA":
+    def _handle_msg_config(self, key: dict, value: Optional[dict]) -> None:
+        subject = key.get("subject")
+        if subject is not None:
+            if subject not in self.subjects:
+                self.log.info("Adding first version of subject: %r with no schemas", subject)
+                self.subjects[subject] = {"schemas": {}}
+
             if not value:
-                subject, version = key["subject"], key["version"]
-                self.log.info("Deleting subject: %r version: %r completely", subject, version)
-                if subject not in self.subjects:
-                    self.log.error("Subject %s did not exist, should have", subject)
-                elif version not in self.subjects[subject]["schemas"]:
-                    self.log.error("Version %d for subject %s did not exist, should have", version, subject)
-                else:
-                    self.subjects[subject]["schemas"].pop(version, None)
-                return
-            schema_type = value.get("schemaType", "AVRO")
-            schema_str = value["schema"]
-            if schema_type in ["AVRO", "JSON"]:
-                try:
-                    ujson.loads(schema_str)  # Validate input json
-                except ValueError:
-                    self.log.error("Invalid json: %s", value["schema"])
-                    return
-                typed_schema = TypedSchema(schema_type=SchemaType(schema_type), schema_str=schema_str)
-            elif schema_type == "PROTOBUF":
-                typed_schema = TypedSchema(schema_type=SchemaType(schema_type), schema_str=schema_str)
+                self.log.info("Deleting compatibility config completely for subject: %r", subject)
+                self.subjects[subject].pop("compatibility", None)
             else:
-                self.log.error("Invalid schema type: %s", schema_type)
-            self.log.debug("Got typed schema %r", typed_schema)
-            subject = value["subject"]
-            if subject not in self.subjects:
-                self.log.info("Adding first version of subject: %r, value: %r", subject, value)
-                self.subjects[subject] = {
-                    "schemas": {
-                        value["version"]: {
-                            "schema": typed_schema,
-                            "version": value["version"],
-                            "id": value["id"],
-                            "deleted": value.get("deleted", False),
-                        }
-                    }
-                }
-                self.log.info("Setting schema_id: %r with schema: %r", value["id"], typed_schema)
-                self.schemas[value["id"]] = typed_schema
-                if value["id"] > self.global_schema_id:  # Not an existing schema
-                    self.global_schema_id = value["id"]
-            elif value.get("deleted", False) is True:
```

Comment:
There is another bug here wrt to the `global_schema_id`. It is the same as the comment above:

https://github.com/aiven/karapace/pull/375/files#r839827344

But here we do have the data to update the global id. The bug is that it was not updated.